### PR TITLE
ShuffleBoard

### PR DESCRIPTION
**Pull request by julianallchin**
- Add shuffleboard support

- Comments and cleanup

- Minor bug fixes to `limelight.py`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .vscode/
 tests/
 __pycache__/
+/tools/
```
Comment (julianallchin) [inline .gitignore:4]: Not necessary

File: limelight.py [hunk 2]
```diff
@@ -6,43 +6,61 @@ class LimeLight:
...
+        current = self.zeroer(self.table.getNumber('ledMode', None))
+
+        if current != 1:
+            self.setLedMode(1)
+        else:
+            self.setLedMode(3)
+
     # Sets limelight’s operation mode
+    def setCamMode(self, value):
         """Sets the camera mode"""
```
Comment (julianallchin) [inline limelight.py]: I guess I just forgot the function definition

File: colorSensorV3.py [hunk 3]
```diff
@@ -192,6 +192,6 @@ def __init__(self, r, g, b):
 
 class ColorSensorV3Sim(i2c_helpers.I2CSimBase):
     def readI2C(self, port, deviceAddress, buffer, count):
-        return len(buffer)
+        return -62
```
Comment (zoecarver) [inline colorSensorV3.py]: What's this number? Add a comment.

File: components/neuralNetwork.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov 14 19:38:54 2019
+@author: Julian Allchin
```
Comment (zoecarver) [inline components/neuralNetwork.py]: Say what this does.

File: components/neuralNetwork.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+"""
+
+import numpy as np
+
+"""ACTIVATION Functions"""
+# Different activation function and its derivative
+def sigmoid(x):
+    return (1/(1+np.exp(-x)))
```
Comment (zoecarver) [inline components/neuralNetwork.py]: Spaces please.

File: components/neuralNetwork.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov 14 19:38:54 2019
+@author: Julian Allchin
+"""
+
+import numpy as np
+
+"""ACTIVATION Functions"""
```
Comment (zoecarver) [inline components/neuralNetwork.py]: Why are there two types of comments here?

File: components/neuralNetwork.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+# Different activation function and its derivative
+def sigmoid(x):
+    return (1/(1+np.exp(-x)))
+def derivativeSigmoid(z):
+     sig=sigmoid(z)
```
Comment (zoecarver) [inline components/neuralNetwork.py]: Spaces please

File: components/neuralNetwork.py [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+     return (sig * (1 - sig))
+ 
+""" COST Functions"""
+# quadratic cost function and its derivative
```
Comment (zoecarver) [inline components/neuralNetwork.py]: Same as above

**Review by zoecarver — CHANGES_REQUESTED** [review]
This looks like it should be three different PRs. The most important thing is that we fix master. Please make a quick patch to do that so I can approve it and we can get that in. Then you can rebase this off master. 

Also, `NeuralNetwork` should be another patch and probably shouldn't be in master until we're actually using it.

**Comment by zoecarver** [comment]
Fixed master in b1b37f2 so you don't have to do that.

**Pull request closed without merging**